src/fabric.c

```diff
@@ -437,42 +437,79 @@ static struct fi_provider *ofi_get_hook(const char *name)
 	return provider;
 }
 
+struct load_entry {
+	const char* provider_name;
+	int should_load;
+};
+
+#define DEFINE_LOAD_ENTRY(NAME, SHOULD_LOAD) \
+    { \
+        .provider_name = NAME, \
+        .should_load = SHOULD_LOAD \
+    }
+
 /* This is the default order that providers will be accessed when available.
  * This, in turn, sets the default ordering of fi_info's reported to the user.
  * However, ofi_reorder_info() may re-arrange the list based on hard-coded
  * criteria.
  */
 static void ofi_ordered_provs_init(void)
 {
-	char *ordered_prov_names[] = {
-		"efa", "psm2", "opx", "usnic", "gni", "bgq", "verbs",
-		"netdir", "psm3", "ucx", "ofi_rxm", "ofi_rxd", "shm",
-
+	struct load_entry ordered_load_list[] = {
+		DEFINE_LOAD_ENTRY("efa", (HAVE_EFA | HAVE_EFA_DL)),
+		DEFINE_LOAD_ENTRY("psm2", (HAVE_PSM2 | HAVE_PSM2_DL)),
+		DEFINE_LOAD_ENTRY("opx", (HAVE_OPX | HAVE_OPX_DL)),
```

Comment:
Window build is broken becasue many of these `HAVE_*` definitions are missing. Need to update `include/windows/config.h`.

Reply:
Which build is the windows build, and if it isn't public -- what are the errors?

Reply:
https://ci.appveyor.com/project/ofiwg/libfabric/builds/48374739 is windows build

Reply:
Thanks @iziemba